code/migration/src/main/scala/com/namely/chiefofstate/migration/versions/v2/MigrateJournal.scala

```diff
@@ -46,36 +53,133 @@ case class MigrateJournal(system: ActorSystem[_], profile: JdbcProfile, serializ
   private val journaldb: JdbcBackend.Database =
     SlickExtension(system).database(system.settings.config.getConfig("jdbc-read-journal")).database
 
-  // get an instance of the default journal dao
-  private val defaultJournalDao: DefaultJournalDao =
-    new DefaultJournalDao(journaldb, profile, journalConfig, serialization)
-
   private val queries: ReadJournalQueries = new ReadJournalQueries(profile, readJournalConfig)
   private val serializer: ByteArrayJournalSerializer =
     new ByteArrayJournalSerializer(serialization, readJournalConfig.pluginConfig.tagSeparator)
 
+  private val newJournalQueries: JournalQueries =
+    new JournalQueries(profile, journalConfig.eventJournalTableConfiguration, journalConfig.eventTagTableConfiguration)
+
   /**
    * write all legacy events into the new journal tables applying the proper serialization
    */
   def run(): Unit = {
-    val source: Source[Try[(PersistentRepr, Set[String], Long)], NotUsed] = Source
-      .fromPublisher(
-        journaldb.stream(queries.JournalTable.result)
-      )
+    val future: Future[Done] = Source
+      .fromPublisher(journaldb.stream(queries.JournalTable.sortBy(_.ordering).result))
       .via(serializer.deserializeFlow)
+      .mapAsync(1)(reprAndOrdNr => Future.fromTry(reprAndOrdNr))
+      .map { case (repr, tags, ordering) => repr.withPayload(Tagged(repr, tags)) -> ordering }
+      .mapAsync(1) { case (repr, ordering) =>
+        // serializing the PersistentRepr using the same ordering received from the old journal
+        val (row, tags): (JournalAkkaSerializationRow, Set[String]) = serialize(repr, ordering)
+        // persist the data
+        writeJournalRows(row, tags)
+      }
+      .runWith(Sink.ignore)
+
+    // run the data migration
+    Await.result(future, Duration.Inf)
+
+    // set the next ordering value
+    log.info("resetting the ChiefOfState new journal ordering sequence number")
+    setNextOrderingValue()
+  }
+
+  /**
+   * returns the next ordering value
+   */
+  private def nextOrderingValue(): Long = {
+    val schemaName = journalConfig.journalTableConfiguration.schemaName.getOrElse(defaultSchemaName)
+    val sequenceName = s"$schemaName.journal_ordering_seq"
+    val stmt: DBIO[Long] =
+      sql"""
+             SELECT pg_catalog.nextval($sequenceName);
+        """.as[Long].head
+
+    Await.result(journaldb.run(stmt), Duration.Inf)
+  }
+
+  /**
+   *  sets the next ordering value
+   */
+  private def setNextOrderingValue(): Long = {
+    val schemaName = journalConfig.eventJournalTableConfiguration.schemaName.getOrElse(defaultSchemaName)
+    val sequenceName = s"$schemaName.event_journal_ordering_seq"
+    val nextVal = nextOrderingValue()
```

Comment:
Yeah but the following line set just the value because if the old journal ordering is 9 the new one after migration should be 10 which is the next value of the old journal ordering.

Reply:
Ah I missed you read it from the old one. Cool.